lib/tasks/assign_compound_sample_supplier_name.rake

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+namespace :compound_sample do
+  desc 'Set the supplier_name on compound samples if all component samples share the same supplier_name'
+  task set_consistent_supplier_name_for_compound_samples: :environment do
```

Comment:
The criteria says " (just for 007c compound samples)". This rake task however does not have any filter to apply only to relevant compound samples. There should be a submission/request based filter I think.